### DISCUSSION TITLE
[Suggestion] Scanners

### DISCUSSION THREAD

**Discussion by PopkaAkula** [tsunamayo/Starship-EVO#4006] [Suggestions]
### The Scanifesto
**An introduction**
When discussing the idea of adding differing armor with some members of the community, we had come across the problem of how to implement various amour textures. Having armors physically look different would encourage sandbox players to use armor decoratively whereas survival players would use it out of necessity, thus creating a disparity between the two. This is one of few problems that could be solved by the Scanner.
### The Scanner
Instead of each armor block having its own texture, you can have a system block that gives you the ability to identify the armor. The scanner would give you information about the object you target and display the information as either a window or overlaid on top of the ship you have scanned. In addition the scanner can also have it’s behavior be changed with upgrades (this will be touched on later). Types of scanners that can exist are:

- **Ship System Scanner**

This would be able to identify and display various ship systems, on the targeted ship, it can show where fuel tanks are, where the reactors are, thrusters, etc. It will be kind of like an x-ray vision but only for ship systems.

- **Armor Scanner**

Sort of like the ship system scanner but oriented more towards identifying weak spots in armor. This would show hull blocks in different colors, (similar to the blue overlay that pipes already have). As an example: light armored hull in green, medium in orange, and heavy in red. 

- **Resource Scanner**

The resource scanner is more of a generalized idea of a scanner that can scan celestial bodies for resources or characteristics, this could potentially be split up into different scanners or kept as one. Some ideas are: a scanner that can detect what biome and temperature a planet is, a scanner that can detect resources on planets and asteroids.

### Scanner Additions

- **Antennas**

Necessary for scanners to work, you can build an antenna to a specific size depending on the specifications you need (scanning radius for example), antennas could be directional, requiring it to be pointing at something in order for an object to be scanned. This would add in a gameplay element of building rotating antennas or placing antennas in specific locations on your ship to make scanners multidirectional. 

- **Holographic projections**

You could link holograms to scanners and receive a holographic projection of what you scanned, this could help with specifically locating things on your scanned target and could play into resource finding.

- **Upgrades**

A basic scanner block would allow players to see the ships that they lock onto. This behavior can be augmented with a system that is already in place: upgrade chips. Upgrade chips for weapons are already in the game, and this could work well with the scanner block. As previously mentioned, the scanner’s behavior (system scanner, resource scanner, etc) could be decided by the chips. This idea could be taken further with chips to perhaps decrease scanning cooldown times, or decrease power consumption. A larger scanner could allow for more potential upgrade chips, but potentially with some drawbacks: greater heat production, and/or electrical consumption.

**Comment by Garrett-C** [comment]
This is a neat idea, someone should mention this on Dev Asks: Killing voxels

**Comment by Garrett-C** [comment]
So just some questions to clarify a few things.

1. I assume the scanner is used to scan an enemy ship, not your own ship yes?
2. Would it not be an idea for the enemy to have some way to counter scanners? Like maybe there is a block designed to dampen scanners or something that you could maybe use to surround a system like a fuel tank to obscure it from a scanner. Maybe even a bit of counter play to this with different thicknesses of the material being more effective and different upgrades to the scanner making it more effective etc.

**Comment by PopkaAkula** [comment]
> So just some questions to clarify a few things.
> 
> 1. I assume the scanner is used to scan an enemy ship, not your own ship yes?
> 2. Would it not be an idea for the enemy to have some way to counter scanners? Like maybe there is a block designed to dampen scanners or something that you could maybe use to surround a system like a fuel tank to obscure it from a scanner. Maybe even a bit of counter play to this with different thicknesses of the material being more effective and different upgrades to the scanner making it more effective etc.

1. I mean realistically you would be able to scan anything but the craft the scanners are on.
2. This could work, however I suggest you open up a separate ticket about counter-scanners if you want to talk about that. 😎😎

**Comment by Garrett-C** [comment]
> > So just some questions to clarify a few things.
> > 
> > 1. I assume the scanner is used to scan an enemy ship, not your own ship yes?
> > 2. Would it not be an idea for the enemy to have some way to counter scanners? Like maybe there is a block designed to dampen scanners or something that you could maybe use to surround a system like a fuel tank to obscure it from a scanner. Maybe even a bit of counter play to this with different thicknesses of the material being more effective and different upgrades to the scanner making it more effective etc.
> 
> 1. I mean realistically you would be able to scan anything but the craft the scanners are on.
> 2. This could work, however I suggest you open up a separate ticket about counter-scanners if you want to talk about that. 😎😎

I would argue that counter scanners are important to keep in mind while thinking about scanners. 😎😎😎

**Comment by ProPeach** [comment]
Solid write up for solid idea 👌

**Comment by PopkaAkula** [comment]
> Solid write up for solid idea 👌

😗👌

**Comment by ZachZent** [comment]
https://github.com/tsunamayo/Starship-EVO/issues/1355
Just going to plug my old scanning idea here as it actually meshes quite well with yours.

**Comment by Kaiser-Indrasil** [comment]
One way you could display armor thickness on a given ship is the World of Warships way:

![WorldOfWarships64_C6kUlRUnCh](https://user-images.githubusercontent.com/63756057/111541607-9c167c80-8768-11eb-87b8-9179f4527df6.jpg)

Another way would be to fade away the rest of the scanned ship's hull whilst focusing on one select block, the Avorion way:

![Avorion_BMBF7pWrZx](https://user-images.githubusercontent.com/63756057/111541696-b6505a80-8768-11eb-9c97-263dabe1e028.png)
![Avorion_uEUdBHXoPk](https://user-images.githubusercontent.com/63756057/111541708-b81a1e00-8768-11eb-992b-94062d8054eb.png)

**Comment by tsunamayo** [comment]
Hi, sorry I would read this when I come to scanner, I have already a lot to catch-up ...
